app/utils/queryBuilder.ts

```diff
@@ -299,6 +133,51 @@ const getWhereExp = (filterGroups: FilterGroup[]): string => {
   return andExps.length > 0 ? `{ _and: [ ${andExps.join("\n")} ]}` : "{}";
 };
 
+/**
+ * Recursively stringify a field tree object into a nested graphql
+ *  field selection set
```

Comment:
you spoiled me with the sample input and output in the function below's docstring, now i want it everywhere 😬